iina/MainWindowController.swift

```diff
@@ -1114,21 +1114,33 @@ class MainWindowController: PlayerWindowController {
         }
       }
     case .windowSize:
-      if fsState.isFullscreen { return }
+      if fsState.isFullscreen {
+        if recognizer.magnification < 0 {
+          self.toggleWindowFullScreen()
+          return
+        }
+      }
 
       // adjust window size
       if recognizer.state == .began {
         // began
         lastMagnification = recognizer.magnification
+        if window.frame.height >= screenFrame.height && lastMagnification > 0 {
+          self.toggleWindowFullScreen()
+          return
+        }
       } else if recognizer.state == .changed {
         // changed
-        let offset = recognizer.magnification - lastMagnification + 1.0;
+        let offset = recognizer.magnification - lastMagnification + 1.0
         let newWidth = window.frame.width * offset
         let newHeight = newWidth / window.aspectRatio.aspect
 
         //Check against max & min threshold
         if newHeight < screenFrame.height && newHeight > minSize.height && newWidth > minSize.width {
-          let newSize = NSSize(width: newWidth, height: newHeight);
+          let newSize = NSSize(width: newWidth, height: newHeight)
+          window.setFrame(window.frame.centeredResize(to: newSize), display: true)
+        } else if newHeight >= screenFrame.height {
+          let newSize = NSSize(width: screenFrame.width, height: screenFrame.height)
```

Comment:
Can just delete this last `else if`. It will end up forcing the window to expand to fill both the width and height of the screen. But doing that will not honor the video's aspect ratio, which is not in line with how IINA currently works.